src/main/java/de/filefighter/rest/domain/filesystem/business/FileSystemBusinessService.java

```diff
@@ -217,190 +215,128 @@ public boolean recursivelyDeleteFileSystemEntity(FileSystemEntity parentFileSyst
+                // check if a file already exists with the same name.
+                // check for children with the name to be uploaded
+                // only necessary when no new folder was created.
+                if (paths.size() == 1) {
+                    List<FileSystemEntity> childrenOfTheParentEntity = fileSystemHelperService.getFolderContentsOfEntityAndPermissions(parentFileSystemEntity, authenticatedUser, false, false);
+                    Optional<FileSystemEntity> optionalEntityWithTheSameName = childrenOfTheParentEntity.stream().filter(children -> children.getName().equals(fileSystemUpload.getName())).findAny();
+                    if (optionalEntityWithTheSameName.isPresent())
+                        throw new FileSystemItemsCouldNotBeUploadedException("A file with the same name already exists.");
+                }
 
-        // user got the item shared.
-        for (long userId : fileSystemEntity.getEditableForUserIds()) {
-            if (userId == authenticatedUser.getUserId())
-                return true;
-        }
+                // create new file
+                uploadedFile = FileSystemEntity.builder()
+                        .fileSystemId(nextId)
+                        .name(fileSystemUpload.getName())
+                        .path(null)
+                        .typeId(fileSystemTypeRepository.parseMimeType(fileSystemUpload.getMimeType()).getId())
+                        .mimeType(fileSystemUpload.getMimeType())
+                        .size(fileSystemUpload.getSize())
+                        .lastUpdated(timeStamp)
+                        .isFile(true)
+                        .createdByUserId(authenticatedUser.getUserId())
+                        .ownerIds(newOwnerIds)
+                        .visibleForUserIds(parentFileSystemEntity.getVisibleForUserIds())
+                        .visibleForGroupIds(parentFileSystemEntity.getVisibleForGroupIds())
+                        .editableForUserIds(parentFileSystemEntity.getEditableForUserIds())
+                        .editableFoGroupIds(parentFileSystemEntity.getEditableFoGroupIds())
+                        .build();
+
+                // set newItemId if not already set
+                if (newItemId == -1) {
+                    newItemId = nextId;
+                }
 
-        // user is in group that got the item shared.
-        long[] fileIsSharedToGroups = fileSystemEntity.getEditableFoGroupIds();
-        for (Group group : authenticatedUser.getGroups()) {
-            for (long groupId : fileIsSharedToGroups) {
-                if (groupId == group.getGroupId())
-                    return true;
+                log.info("Creating new file {} for user {}.", uploadedFile, authenticatedUser.getUserId());
+                fileSystemRepository.save(uploadedFile);
+            } else {
+                // concat pathToCheck with currentPath
+                folderPathToCheck += folderPathToCheck.equals("/") ? currentPath : "/" + currentPath;
```

Comment:
Does this concat the absolute path of the parent with the relative path of the new child?

Reply:
yes.